src/Sensorario/DiamondKata/DiamondKata.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Sensorario\DiamondKata;
+
+final class DiamondKata
+{
+    const EOL = "\n";
+
+    private $params;
+
+    private function __construct(array $params)
+    {
+        $this->params['input'] = $params['input'];
+    }
+
+    public static function createFrom($input)
+    {
+        return new self([
+            'input' => $input
+        ]);
+    }
```

Comment:
I've changed method signature in ::withChar(). And the reason is that is more readable than __construct. And, ... I like Value Object, so DiamondKata is a value object.